dataflow/src/main/java/org/checkerframework/dataflow/reachingdefinitions/ReachingDefinitionsValue.java

```diff
@@ -0,0 +1,51 @@
+package org.checkerframework.dataflow.reachingdefinitions;
+
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.checkerframework.dataflow.analysis.AbstractValue;
+import org.checkerframework.dataflow.cfg.node.AssignmentNode;
+import org.checkerframework.javacutil.BugInCF;
+
+/**
+ * A reaching definitions (which is represented by a node) wrapper turning node into abstract value.
+ */
+public class ReachingDefinitionsValue implements AbstractValue<ReachingDefinitionsValue> {
+
+    /**
+     * A reaching definition is represented by a node, which can be a {@link
+     * org.checkerframework.dataflow.cfg.node.AssignmentNode}.
+     */
+    protected final AssignmentNode def;
+
+    @Override
+    public ReachingDefinitionsValue leastUpperBound(ReachingDefinitionsValue other) {
+        throw new BugInCF("lub of reachDef get called!");
+    }
+
+    /**
+     * Create a new definition.
+     *
+     * @param n a node
+     */
+    public ReachingDefinitionsValue(AssignmentNode n) {
+        this.def = n;
+    }
+
+    @Override
+    public int hashCode() {
+        return this.def.hashCode();
+    }
+
+    @Override
+    public boolean equals(@Nullable Object obj) {
+        if (!(obj instanceof ReachingDefinitionsValue)) {
+            return false;
+        }
+        ReachingDefinitionsValue other = (ReachingDefinitionsValue) obj;
+        return this.def.equals(other.def);
```

Comment:
Not using `toString` here so I'd assume `equals` works as expected.

Reply:
Yeah, works well here.

Reply:
Nodes should not be compared with `.equals`, always use reference equality `==`.

Reply:
Thanks! Updated.